### DISCUSSION TITLE
can velero support app migration between different version of  kubernetes clusters

### DISCUSSION THREAD

**Discussion by Crazy-Kitty** [vmware-tanzu/velero#3265] [Community support Q&A] [answered]
**Describe the problem/challenge you have**
[A description of the current limitation/problem/challenge that you are experiencing.]
I deploy some application in k8s cluster which version is 1.12. And I want to migrate the application to new a k8s cluster which version is 1.18 or more. But If the same application runs on new k8s cluster, it will have some errors such as 'no matches for kind "Deployment" in version "extensions/v1beta1"'. Can velero help resolve the problem if I backup the application with velero in old version k8s cluster and restore it in new version k8s cluster?

**Describe the solution you'd like**
[A clear and concise description of what you want to happen.]

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- Velero version (use `velero version`): 1.5.2
- Kubernetes version (use `kubectl version`): 1.12/1.18
- Kubernetes installer & version:
- Cloud provider or hardware configuration:
- OS (e.g. from `/etc/os-release`):

**Vote on this issue!**

This is an invitation to the Velero community to vote on issues, you can see the project's [top voted issues listed here](https://github.com/vmware-tanzu/velero/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc).  
Use the "reaction smiley face" up to the right of this comment to vote.

- :+1: for "The project would be better with this feature added"
- :-1: for "This feature will not enhance the project in a meaningful way"

**Comment by codegold79** [comment, marked as answer]
There is a new feature (scheduled to be released with Velero 1.6) that allows for skipping Kubernetes versions (to a certain extent, and with some limitations). The feature is called Enable API Group Versions and the related PR is #3133. Included with that PR are updates to the Velero documentation website that explain the feature.

Let me know if you have more questions.

**Comment by Crazy-Kitty** [comment]
thanks a lot. It is great
